flink-runtime/src/main/java/org/apache/flink/runtime/metrics/util/MetricUtils.java

```diff
@@ -131,11 +147,47 @@ public static void instantiateStatusMetrics(
 
 		instantiateClassLoaderMetrics(jvm.addGroup("ClassLoader"));
 		instantiateGarbageCollectorMetrics(jvm.addGroup("GarbageCollector"));
-		instantiateMemoryMetrics(jvm.addGroup("Memory"));
+		instantiateMemoryMetrics(jvm.addGroup(METRIC_GROUP_MEMORY));
 		instantiateThreadMetrics(jvm.addGroup("Threads"));
 		instantiateCPUMetrics(jvm.addGroup("CPU"));
 	}
 
+	public static void instantiateFlinkMemoryMetricGroup(
+			MetricGroup parentMetricGroup,
+			TaskManagerServices taskManagerServices) {
```

Comment:
nit: I think `TaskManagerServices` is too heavy to pass for managed memory metrics.
I would pass `TaskSlotTable` and managed memory size.